tko_web_sessions_management/main.py

```diff
@@ -75,6 +75,9 @@ def web_login(self, redirect=None, **kw):
             if 'login' in request.params and 'password' in request.params:
                 uid = request.session.authenticate(request.session.db, request.params[
                                                    'login'], request.params['password'])
+            reason1 = False
+            reason2 = False
+            reason3 = False
```

Comment:
Please don't say to user why he can't access that is a security breach.
By displaying exactly what is wrong with the login attempt, you are helping an attacker narrow down his attack. We should give always a generic message.
Although we log the reason in user attempts so admin or support can explain or fix the problem.